src/ansys/dpf/composites/ply_wise_data.py

```diff
@@ -0,0 +1,87 @@
+"""Methods to get ply-wise data from a result field."""
+
+from enum import Enum, IntEnum
+from typing import Union
+
+from ansys.dpf.core import Field, MeshedRegion, Operator, operators
+from ansys.dpf.gate.common import locations
+
+__all__ = ("ReductionStrategy", "get_ply_wise_data")
+
+
+class ReductionStrategy(Enum):
+    """The reduction strategy to get from spot values (BOT, MID, TOP) to a single value."""
+
+    MIN = "MIN"
+    MAX = "MAX"
+    AVG = "AVG"
+    BOT = "BOT"
+    MID = "MID"
+    TOP = "TOP"
+
+
+def get_ply_wise_data(
+    field: Field,
+    ply_name: str,
+    mesh: MeshedRegion,
+    reduction_strategy: ReductionStrategy = ReductionStrategy.AVG,
+    requested_location: str = locations.elemental_nodal,
+    component: Union[IntEnum, int] = 0,
+) -> Field:
+    """Get ply-wise data from a field.
+
+    Parameters
+    ----------
+    field:
+        Field to extract data from.
+    ply_name:
+        Name of the ply to extract data from.
+    mesh :
+        Meshed region. Needs to be enriched with composite information.
```

Comment:
```suggestion
        Meshed region to enrich with composite information.
```
I redid this because "Needs to be enriched...." isn't a complete sentence on its own. Please revise if this isn't what was meant.